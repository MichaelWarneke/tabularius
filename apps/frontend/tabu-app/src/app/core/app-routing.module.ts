import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  {
    path: 'dash',
    loadChildren: '@tabularius/modules/app-dash#AppDashModule'
  },
  {
    path: 'journal',
    loadChildren: '@tabularius/modules/app-journal#AppJournalModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: []
})
export class AppRoutingModule {}
