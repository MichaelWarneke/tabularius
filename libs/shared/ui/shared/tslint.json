{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
