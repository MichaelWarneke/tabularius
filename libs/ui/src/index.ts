export { TabuAppSidenavModule } from './lib/nav';
