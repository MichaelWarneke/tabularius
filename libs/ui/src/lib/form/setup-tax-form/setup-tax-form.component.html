<h1>Tax Form</h1>
<tabu-dynamic-form
  [formModel]="formModel"
  [data]="_user"
  (saveEntry)="onSave($event)"
></tabu-dynamic-form>
