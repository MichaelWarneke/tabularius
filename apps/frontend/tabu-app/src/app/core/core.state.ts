import {
  ActionReducerMap,
  MetaReducer,
  createFeatureSelector
} from '@ngrx/store';
import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { RouterStateUrl } from '@tabularius/shared/services';
import { storeFreeze } from 'ngrx-store-freeze';

import { initStateFromLocalStorageReducer } from '@tabularius/shared/services';
import { debugReducer } from '@tabularius/shared/services';

import { environment } from '../../environments/environment';

export interface AppState {
  router: RouterReducerState<RouterStateUrl>;
}

export const reducers: ActionReducerMap<AppState> = {
  router: routerReducer
};

export const metaReducers: MetaReducer<AppState>[] = [
  initStateFromLocalStorageReducer
];
if (!environment.production) {
  metaReducers.unshift(storeFreeze);
  metaReducers.unshift(debugReducer);
}

export const selectRouterState = createFeatureSelector<
  AppState,
  RouterReducerState<RouterStateUrl>
>('router');
