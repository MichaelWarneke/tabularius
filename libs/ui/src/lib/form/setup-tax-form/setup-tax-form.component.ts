import {
  Component,
  OnInit,
  Output,
  EventEmitter,
  Input,
  ChangeDetectionStrategy
} from '@angular/core';
import { ITax } from '@tabularius/shared/models';
import { QuestionService } from './services/form-creator.service';

@Component({
  selector: 'tabu-setup-tax-form',
  templateUrl: './setup-tax-form.component.html',
  styleUrls: ['./setup-tax-form.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: [QuestionService]
})
export class SetupTaxFormComponent implements OnInit {
  @Output() saveTax = new EventEmitter<ITax>();
  @Output() deleteTax = new EventEmitter<string>();
  @Input() tax: ITax | null = null;

  questions: any[];

  constructor(service: QuestionService) {
    this.questions = service.getQuestions();
  }

  ngOnInit() {}

  onCopy() {}

  onDelete() {}

  onSubmit() {
    console.log('Submit clicked');
  }
}
