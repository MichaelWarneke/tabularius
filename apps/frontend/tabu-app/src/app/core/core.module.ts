import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppRoutingModule } from './app-routing.module';
import { TabuAppSidenavModule } from '@tabularius/ui';

@NgModule({
  declarations: [],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    TabuAppSidenavModule
  ],
  exports: [TabuAppSidenavModule]
})
export class CoreModule {}
