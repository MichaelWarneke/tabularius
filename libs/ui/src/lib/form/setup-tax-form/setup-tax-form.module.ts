import { NgModule } from '@angular/core';
import { SharedUiModule, DynamicFormModule } from '@tabularius/shared/ui';
import { SetupTaxFormComponent } from './setup-tax-form.component';

@NgModule({
  declarations: [SetupTaxFormComponent],
  imports: [SharedUiModule, DynamicFormModule],
  exports: [SetupTaxFormComponent]
})
export class SetupTaxFormModule {}
