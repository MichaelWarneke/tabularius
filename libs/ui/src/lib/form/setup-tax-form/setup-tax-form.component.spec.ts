import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

import { SetupTaxFormComponent } from './setup-tax-form.component';
import { SharedModule } from '@tabularius/shared/ui';

describe('SetupTaxFormComponent', () => {
  let component: SetupTaxFormComponent;
  let fixture: ComponentFixture<SetupTaxFormComponent>;
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [NoopAnimationsModule, SharedModule],
      declarations: [SetupTaxFormComponent],
      providers: []
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SetupTaxFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
