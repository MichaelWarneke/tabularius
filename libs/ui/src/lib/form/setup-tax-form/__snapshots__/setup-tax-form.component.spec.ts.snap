// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupTaxFormComponent should render 1`] = `
<tabu-setup-tax-form
  deleteTax={[Function EventEmitter]}
  formModel={[Function FormSetup]}
  saveTax={[Function EventEmitter]}
  tax="null"
>
  <h1>
    Tax Form
  </h1><tabu-dynamic-form />
</tabu-setup-tax-form>
`;
